### DISCUSSION TITLE
Facing issue in GET , SET & ADD device API.

### DISCUSSION THREAD

**Discussion by praveen2211** [OktopUSP/oktopus#102] [General]
@leandrofars  I am able to bring up the copde and register device with USP but facing issue while GET and  SET end points.
Request URL -
Response in postman for GET - api/device/os::4851CF-000000000002/get

{
    "req_path_results": [
        {
            "requested_path": "Device.WiFi.SSID.[Name==wlan0].",
            "err_code": 7016,
            "err_msg": "DM_PRIV_GetNodeFromPath: Path is invalid: Device.WiFi.SSID."
        },
        {
            "requested_path": "Device.IP.Interface.*.Alias",
            "err_code": 7016,
            "err_msg": "DM_PRIV_GetNodeFromPath: Path is invalid: Device.IP.Interface."
        },
        {
            "requested_path": "Device.DeviceInfo.FirmwareImage.*.Alias",
            "err_code": 7016,
            "err_msg": "DM_PRIV_GetNodeFromPath: Path is invalid: Device.DeviceInfo.FirmwareImage."
        },
        {
            "requested_path": "Device.IP.Interface.1.IPv4Address.1.IPAddress",
            "err_code": 7026,
            "err_msg": "CheckPathProperties: Path (Device.IP.Interface.1.IPv4Address.1.IPAddress) does not exist in the schema"
        }
    ]
}

Error while performing SET operation

Request - /api/device/os::4851CF-000000000002/set
Response  -
{
    "updated_obj_results": [
        {
            "requested_path": "Device.IP.Interface.[Alias==pamonha].",
            "oper_status": {
                "OperStatus": {
                    "OperFailure": {
                        "err_code": 7016,
                        "err_msg": "DM_PRIV_GetNodeFromPath: Path is invalid: Device.IP.Interface."
                    }
                }
            }
        }
    ]
}

**Comment by leandrofars** [comment]
Your device appears at frontend devices list?

**Reply by praveen2211** [reply]
Hi @leandrofars ,

Yes device appears at frontend .

![Device-Online](https://github.com/OktopUSP/oktopus/assets/6102137/1526e119-8bb9-47a5-8611-14288f8ed4f8)

**Comment by praveen2211** [comment]
Please find Response for ADD end point 

{
    "created_obj_results": [
        {
            "requested_path": "Device.IP.Interface.",
            "oper_status": {
                "OperStatus": {
                    "OperFailure": {
                        "err_code": 7026,
                        "err_msg": "CheckPathProperties: Path (Device.IP.Interface) does not exist in the schema"
                    }
                }
            }
        }
    ]
}

**Reply by leandrofars** [reply]
It means your device doesn't have those parameters (Device.IP.Interface) implemented at the data model

**Reply by leandrofars** [reply]
I think you're just running obuspa right? but it's not running in a real device and doesn't have the vendor specific functions implemented, so you'll not be able to fetch those parameters properly.
